File: day12/word_count.py
# Every once in awhile I remember to create functions and __main__ !!!

def count_words():
    while True:
        sentence = input("Please type a sentence (one or more words): ").strip()
        if not sentence:
            print("Input cannot be empty.\n")
            continue

        words = sentence.split()
        count = len(words)

        print(f"There {'is' if count == 1 else 'are'} {count} word{'s' if count != 1 else ''} in your sentence.")
        break

if __name__ == "__main__":
    count_words()

File: day12/WordCount.java
import java.util.Scanner;

public class WordCount {
    public static void main(String[] args) {
        countWords();
    }

    public static void countWords() {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print("Please type a sentence (one or more words): ");
            String input = scanner.nextLine().trim();

            if (input.isEmpty()) {
                System.out.println("Input cannot be empty.\n");
                continue;
            }

            String[] words = input.split("\\s+");
            int count = words.length;

            if (count == 1) {
                System.out.println("There is 1 word in your sentence.");
            } else {
                System.out.println("There are " + count + " words in your sentence.");
            }
            break;
        }

        scanner.close();
    }
}
